be/src/agent/agent_server.cpp

```diff
@@ -110,25 +116,43 @@ void AgentServer::Impl::init_or_die() {
         }
     }
 
-    auto st =
-            ThreadPoolBuilder("publish_version")
-                    .set_min_threads(config::transaction_publish_version_worker_count)
-                    .set_max_threads(config::transaction_publish_version_worker_count)
-                    // The ideal queue size of threadpool should be larger than the maximum number of tablet of a partition.
-                    // But it seems that there's no limit for the number of tablets of a partition.
-                    // Since a large queue size brings a little overhead, a big one is chosen here.
-                    .set_max_queue_size(2048)
-                    .build(&_thread_pool_publish_version);
-    CHECK(st.ok()) << st;
+#define BUILD_DYNAMIC_TASK_THREAD_POOL(name, min_threads, max_threads, queue_size, pool) \
```

Comment:
why not use a function? I'm afraid the macro is not suitable and not necessary?

Reply:
prefer an inline func, but the original code uses micro, like `CREATE_AND_START_POOL()`, so here just follow the tradition